### PR TITLE
[PB-5925]: add native InternxtApiClient for DocumentsProvider

### PR DESCRIPTION
**Pull request by terrerox**
Adds a pure-Kotlin HTTP client that talks directly to the Drive and Bridge APIs so the SAF DocumentsProvider can fetch folder contents and download links without the React Native bridge. Covers list/create/rename/move/trash on Drive (Bearer auth) and getDownloadLinks on Bridge (Basic auth with sha256(userId).hex derivation). Typed errors for 401/404/other/network; MockWebServer tests for parsing, auth headers, gateway-required headers, and error mapping.

### CODE REVIEW

File: android/app/src/main/java/com/internxt/cloud/documents/api/AuthConfig.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.internxt.cloud.documents.api
+
+data class AuthConfig(
+    val driveBaseUrl: String,
+    val bridgeBaseUrl: String,
+    val bearerToken: String,
+    val bridgeUser: String,
+    val userId: String,
+    val clientName: String = "drive-mobile",
+    val clientVersion: String = "v1.9.0",
```
Comment (CandelR) [inline android/app/src/main/java/com/internxt/cloud/documents/api/AuthConfig.kt]: Perhaps we should make this configurable so that there is a single source of truth. Otherwise, we might as well forget about changing it here

File: android/app/src/main/java/com/internxt/cloud/documents/api/InternxtApiClient.kt [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+            .header("internxt-client", config.clientName)
+            .header("internxt-version", config.clientVersion)
+        config.desktopToken?.let { builder.header("x-internxt-desktop-header", it) }
+        return builder
+    }
+
+    private fun driveUrl(path: String) = "${config.driveBaseUrl.trimEnd('/')}/$path".toHttpUrl()
+    private fun bridgeUrl(path: String) = "${config.bridgeBaseUrl.trimEnd('/')}/$path".toHttpUrl()
+
+    private fun execute(request: Request): JSONObject {
```
Comment (CandelR) [inline android/app/src/main/java/com/internxt/cloud/documents/api/InternxtApiClient.kt]: I think that this fun name could be a little more descriptive

File: android/app/src/main/java/com/internxt/cloud/documents/api/InternxtApiClient.kt [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+}
+
+private fun JSONObject.optStringOrNull(key: String): String? =
+    if (isNull(key)) null else optString(key).takeIf { it.isNotEmpty() }
+
+private fun JSONObject.optLongFlexible(key: String): Long = when (val v = opt(key)) {
+    is Number -> v.toLong()
+    is String -> v.toLongOrNull() ?: 0L
+    else -> 0L
+}
```
Comment (CandelR) [inline android/app/src/main/java/com/internxt/cloud/documents/api/InternxtApiClient.kt]: Perhaps I would extract these functions and add some tests

File: android/app/src/main/java/com/internxt/cloud/documents/crypto/HashUtil.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.internxt.cloud.documents.crypto
+
+import java.security.MessageDigest
+
+object HashUtil {
```
Comment (CandelR) [inline android/app/src/main/java/com/internxt/cloud/documents/crypto/HashUtil.kt:5]: add @TamaraFinogina  as codeowner for crypto directory please :)

File: android/app/src/test/java/com/internxt/cloud/documents/api/InternxtApiClientTest.kt [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.SocketPolicy
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertThrows
+import org.junit.Before
+import org.junit.Test
+
+class InternxtApiClientTest {
```
Comment (CandelR) [inline android/app/src/test/java/com/internxt/cloud/documents/api/InternxtApiClientTest.kt:14]: I would add more tests, for example for listFolderFolders, createFolder, null optional fields in parseFile/parseFolder, size as a string value, and 5xx ApiError

File: android/app/src/main/java/com/internxt/cloud/documents/api/InternxtApiClient.kt [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+            .put("parentFolderUuid", parentUuid)
+        val req = driveRequest(driveUrl("folders"))
+            .post(payload.toString().toRequestBody(JSON))
+            .build()
+        return parseFolder(executeApiRequest(req))
+    }
+
+    fun renameFile(fileUuid: String, newName: String): DriveFile {
+        val payload = JSONObject().put("name", newName)
+        val req = driveRequest(driveUrl("files/$fileUuid"))
```
Comment (CandelR) [inline android/app/src/main/java/com/internxt/cloud/documents/api/InternxtApiClient.kt:63]: I think that the correct endpoint for rename is files/$fileUuid/meta and is a put, not a patch, unless there is a new one. Confirm it please

File: android/app/src/main/java/com/internxt/cloud/documents/api/InternxtApiClient.kt [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    }
+
+    companion object {
+        const val DEFAULT_PAGE_SIZE = 50
+
+        private val JSON = "application/json; charset=utf-8".toMediaType()
+
+        private fun defaultClient(): OkHttpClient = OkHttpClient.Builder()
+            .connectTimeout(30, TimeUnit.SECONDS)
+            .readTimeout(30, TimeUnit.SECONDS)
```
Comment (CandelR) [inline android/app/src/main/java/com/internxt/cloud/documents/api/InternxtApiClient.kt]: I think adding a [writeTimeout](https://square.github.io/okhttp/5.x/okhttp/okhttp3/-ok-http-client/-builder/write-timeout.html) could also save us future problems when uploading files. Also is gonna be the same for all I think we can use [callTimeout](https://square.github.io/okhttp/5.x/okhttp/okhttp3/-ok-http-client/-builder/call-timeout.html?query=fun%20callTimeout(duration:%20Duration):%20OkHttpClient.Builder). What do you think?

File: android/app/src/main/java/com/internxt/cloud/documents/api/InternxtApiClient.kt (hunk 6 above)
Comment (terrerox) [reply android/app/src/main/java/com/internxt/cloud/documents/api/InternxtApiClient.kt:63]: Ive changed it in this pr https://github.com/internxt/drive-mobile/pull/440/changes

**Comment by terrerox** [comment]
@CandelR this one is ready for your review

**Review by CandelR — APPROVED** [review]
